.github/workflows/ci-server.yml

```diff
@@ -0,0 +1,53 @@
+name: CI - Server
+
+on:
+  push:
+    paths:
+      - 'apps/server/**'
+    branches:
+      - main
+  pull_request:
+    types: [opened, synchronize, reopened]
+    paths:
+      - 'apps/server/**'
+    branches:
+      - '**'
+
+jobs:
+  lint:
+    name: Lint & Types
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 18
+
+      - name: Clean install dependencies
+        run: npm ci
+
+      - name: Lint & Check types
+        run: |
+          npm run lint
+          npm run types:check
+        working-directory: ./apps/server
+
+  build:
+    name: Build
+    needs: [lint]
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 18
+
+      - name: Clean install dependencies
+        run: npm ci
+
+      - name: Build
+        run: |
+          npm run build
```

Comment:
Por que executar o `build`? Isso vai ajudar a validar alguma coisa?

Reply:
Eu acho legal rodar o `build` pra garantir que a aplicação continua buildando normalmente com as alterações, pra não ter nenhuma surpresa quando for buildar no deploy.

Mas se vc achar que faz sentido remover pro job executar mais rápido, posso remover

Reply:
Acredito que o único motivo de o build não estar funcionando é não passar algum type checking, não? Não sei se poderia rolar outro problema 🤔

Reply:
:thinking: faz sentido, vou remover